assets/prototype/application/services/entityTextFieldSearch/index.ts

```diff
@@ -0,0 +1,28 @@
+type EntityTextFieldSearchProps = {
+	entities: any[];
+	searchText: string;
+	field?: string;
+};
+
+/**
+ * searches through provided list of entities and returns an array
+ * of entities where the set field (defaults to entity name)
+ * matches the current search text (fully or partially)
+ *
+ * @param {BaseEntity[]} entities
+ * @param {string} searchText
+ * @param {string} field
+ * @return {BaseEntity[]} entities matching search criteria
+ */
+const entityTextFieldSearch = ({ entities, searchText }: EntityTextFieldSearchProps) => {
+	const isSearchable = searchText && Array.isArray(entities);
+	const searchResult = entities.filter(({ name: title }) => {
```

Comment:
As discussed with @tn3rb it will search only through titles on the first iteration.
After proto will add more searchable fields.